src/java.base/share/conf/security/java.security

```diff
@@ -635,6 +635,41 @@ jdk.certpath.disabledAlgorithms=MD2, MD5, SHA1 jdkCA & usage TLSServer, \
     RSA keySize < 1024, DSA keySize < 1024, EC keySize < 224, \
     SHA1 usage SignedJAR & denyAfter 2019-01-01
 
+#
+# Algorithm restrictions for Hybrid Public Key Encryption (HPKE)
+#
+# In some environments, certain algorithm identifiers may be undesirable
+# for Hybrid Public Key Encryption. See the specification of HPKEParameterSpec
+# for definition of algorithm identifiers used in HPKE.
+#
+# The value for this property is a comma-separate-list of equations. Each one
+# taking the form of one of
+#
+#   kem_id=<number>
+#   kem_id=<number>-<number>
+#   kdf_id=<number>
+#   kdf_id=<number>-<number>
+#   aead_id=<number>
+#   aead_id=<number>-<number>
+#
+# If only one number is provided, the algorithm identifier is disabled. If a
+# range is provided, the algorithm identifiers within the range (inclusive
+# on both sides) are disabled. Initializing an HPKE cipher with disabled
+# algorithm identifier(s) will throw an InvalidAlgorithmParameterException.
+#
+# The numbers can be in any format that Integer::decode accepts.
+# All whitespaces around delimiters are ignored. The names are case-sensitive.
+#
+# Note: This property is currently used by the JDK Reference
+# implementation. It is not guaranteed to be examined and used by other
+# implementations.
+#
+# Example:
+#   jdk.hpke.disabledAlgorithms=kem_id=16-18
+#   jdk.hpke.disabledAlgorithms=kem_id=0x10,kdf_id=0x01,aead_id=0xffff
+#
+jdk.hpke.disabledAlgorithms=
```

Comment:
Do you expect that these algorithm ids would be embedded in some higher level network protocols or artifacts and thus could potentially be leveraged that way? Trying to understand the motivation/need for this property, as we have not (as of yet) disabled algorithms at the JCE layer, with the rationale that it is up to the developer to understand the security risks of directly using a weak or crypto algorithm.

There may also be some overlap with Red Hat's proposed Security Providers Filter.

Reply:
You’re right. I can remove this part if there’s no requirement yet. For TLS, we already enforce constraints through the existing TLS security property. Perhaps another usage scope is enough when we start supporting Encrypted ClientHello.

Reply:
Removed.